python/pyarrow/tests/test_scalars.py

```diff
@@ -151,3 +151,23 @@ def test_dictionary(self):
                                            categorical.categories)
         for i, c in enumerate(values):
             assert v[i].as_py() == c
+
+    def test_struct(self):
```

Comment:
This should be split into two tests: one that tests `{from,to}_pylist` in `tests/test_convert_builtin.py` and one that tests `{from,to}_pandas` in `tests/test_convert_pandas.py`.